lib/paper_trail_scrapbook/changes.rb

```diff
@@ -63,17 +71,40 @@ def find_value(key, value)
       return '*empty*' unless value
 
       begin
-        build_associations[key].find(value).to_s.to_s + "[#{value}]"
+        assoc_target(key).find(value).to_s.to_s + "[#{value}]"
       rescue StandardError
         "*not found*[#{value}]"
       end
     end
 
+    def assoc_target(key)
+      x = build_associations[key]
+      return x unless polymorphic?(x)
+      ref = x[1..-1] + '_type'
+
+      # try object changes to see if the belongs_to class is specified
+      latest_class = changes[ref]&.last
+
+      if latest_class.nil? && create?
+        # try the db default class
+        # for creates where the object changes do not specify this it
+        # is most likely because the default ==  type selected so
+        # the default was not changed and therefore is not in
+        # object changes
+        orig_instance = Object.const_get(version.item_type.classify).new
+        latest_class  = orig_instance[ref.to_sym]
+      end
+
+      Object.const_get(latest_class.classify)
+    end
+
     def assoc_klass(name, options = {})
       direct_class = options[:class_name]
-      return direct_class if direct_class && !direct_class.is_a?(String)
+      poly         = options[:polymorphic]
+
+      return direct_class if !poly && direct_class && !direct_class.is_a?(String)
 
-      Object.const_get((direct_class || name.to_s).classify)
+      poly ? '*' + name.to_s : Object.const_get((direct_class || name.to_s).classify)
```

Comment:
This might be a bit cleaner:

```suggestion
      poly ? "*#{name}" : Object.const_get((direct_class || name.to_s).classify)
```

Reply:
Looking at this I should use a constant